source/js/src/even.js

```diff
@@ -106,12 +106,16 @@
 
   Even.prototype.tocFollow = function () {
     var HEADERFIX = 30;
+    var isFirefox = navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
     var $toclink = $('.toc-link'),
       $headerlink = $('.headerlink');
 
     $(window).scroll(function () {
       var scrollTop = $(window).scrollTop();
       var headerlinkTop = $.map($headerlink, function (link) {
+        if(isFirefox) {
+          return $(link).offset().top;
```

Comment:
I believe this is a bug of jQuery

Reply:
Managed to fully understand the issue, and I'll try to explain it here.

The markdown renderer of `hexo-theme-even` will generate these codes for headers:

```html
<h2 id="测试标题">
  <a href="#测试标题" class="headerlink" title="测试标题"></a>
  测试标题
</h2>
```

Since `<a>` tag has no text, it is a transparent link.

Let's have a look at jQuery's `$.prototype.offset`:

```js
offset: function(a) {
  if (arguments.length)
    return void 0 === a ? this : this.each(function(b) {
      r.offset.setOffset(this, a, b)
    });
  var b, c, d, e, f = this[0];
  if (f)
    return f.getClientRects().length ? (d = f.getBoundingClientRect(),
    // (1)
    d.width || d.height ? (e = f.ownerDocument,
    c = Sb(e),
    b = e.documentElement,
    {
      // (2)
      top: d.top + c.pageYOffset - b.clientTop,
      left: d.left + c.pageXOffset - b.clientLeft
    // (3)
    }) : d) : {
      top: 0,
      left: 0
    }
}
```

In Chrome (or let's say, WebKit), this `<a class="headerlink">` is `0*0` pixels big, but in Firefox, it is `0*34.166`, which means that at point `(1)`, `d.width || d.height` evaluates to `false` in WebKit, and `true` in Firefox. Thus, WebKit browsers got to `(3)` and directly returns `d`, while Firefox goes to `(2)` and returns `d.top + c.pageYOffset - b.clientTop`. In this case, `c.pageYOffset` equals to `$(window).scrollTop()`, and that's why WebKit's `.top` = Firefox's `.top - $(window).scrollTop()`.

Maybe I'll write a blog about this later :-)

Reply:
Here comes the final solution: this **is** a bug of jQuery, but it has been fixed in the latest version of jQuery (I've tested that.), so you can either merge my PR to patch it or upgrade your `lib/jquery.min.js`'s version to the latest version.

Have a nice day :-)